app/services/structured_fact_importer.rb

```diff
@@ -19,6 +19,28 @@ def normalize_recurse(memo, facts, prefix = '')
     memo
   end
 
+  def find_or_create_fact_name(name, value)
+    if name.include?(FactName::SEPARATOR)
```

Comment:
This is just refactor of the current create_fact_name (see the second implementation below, the one I added was just a miscopy).

I don't agree with extracting this method, this is a compact recursive method. Codeclimate is simply wrong here. Let me push and show you how it looks like.

I am also adding more benchmarks to test the structured importer to see the difference.